apimachinery/src/request.rs

```diff
@@ -0,0 +1,187 @@
+use crate::meta::GroupVersionResource;
+use failure::Error;
+use http;
+use http::header::{HeaderValue, ACCEPT, CONTENT_TYPE};
+use serde::Serialize;
+use serde_json;
+use serde_urlencoded;
+
+pub const APPLICATION_JSON: &str = "application/json";
+pub const JSON_PATCH: &str = "application/json-patch+json";
+pub const MERGE_PATCH: &str = "application/merge-patch+json";
+pub const STRATEGIC_MERGE_PATCH: &str = "application/strategic-merge-patch+json";
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum Patch {
+    Json(serde_json::Value),
+    Merge(serde_json::Value),
+    StrategicMerge(serde_json::Value),
+}
+
+impl Patch {
+    pub fn content_type(&self) -> &'static str {
+        match self {
+            Patch::Json(_) => JSON_PATCH,
+            Patch::Merge(_) => MERGE_PATCH,
+            Patch::StrategicMerge(_) => STRATEGIC_MERGE_PATCH,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Request<B, O> {
```

Comment:
I think it would be useful to say BODY or Body rather than B here, similar OPTS - we're not running out of space :P